clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -278,14 +303,66 @@ protected void onJoinComplete(int generation,
 
         // execute the user's callback after rebalance
         ConsumerRebalanceListener listener = subscriptions.rebalanceListener();
-        log.info("Setting newly assigned partitions: {}", Utils.join(assignedPartitions, ", "));
+
+        switch (protocol) {
+            case EAGER:
+                if (!ownedPartitions.isEmpty()) {
```

Comment:
Is this a possible state?

Reply:
This is not possible state, and hence I'm logging a WARN. I can also just throw an exception actually.